pin-project-internal/src/pin_project/derive.rs

```diff
@@ -970,19 +973,19 @@ fn make_proj_impl(
         }
     });
 
-    if cx.kind == Enum {
-        if !cx.project {
-            project = None;
-        }
-        if !cx.project_ref {
-            project_ref = None;
-        }
-        if cx.project_replace.ident().is_none() {
-            project_replace = None;
-        }
+    if cx.project != expose {
+        project = None;
+    }
+    if cx.project_ref != expose {
+        project_ref = None;
+    }
+    if cx.project_replace.ident().is_some() != expose {
+        project_replace = None;
     }
 
+    let allowed_lints = global_allowed_lints();
     quote! {
+        #allowed_lints
```

Comment:
This does *not* generate a `#[allow(dead_code)]`, which is also applied globally to the private scope. IMO, renaming the projection is a pretty clear signal that you're planning to use it, and silencing the warning is a false negative in this case.

Reply:
The CI failures are about these lints. Leaving them until a consensus has been reached on where to go about that.

Reply:
Basically, the warning is not issued to the method itself even if no `#[allow(dead_code)]` because `fn project()` has call-site span. The problem is the macro_rules macro changes the span (https://github.com/rust-lang/rust/issues/77973).

I would prefer to use `#[allow(dead_code)]` because it's preferable to work the same as when it's not called inside macros.

Reply:
Yeah makes sense.

Reply:
Alright, tried to get the spans to make sense but couldn't get it to cooperate, so I added the `#[allow]`.